Libraries/Components/View/ViewPropTypes.js

```diff
@@ -100,22 +100,64 @@ type DirectEventProps = $ReadOnly<{|
 |}>;
 
 // [macOS
-type KeyboardEventProps = $ReadOnly<{|
+/**
+ * Represents a key that could be passed to `validKeysDown` and `validKeysUp`.
+ *
+ * `key` is the actual key, such as "a", or one of the special values:
+ * "Tab", "Escape", "Enter", "ArrowLeft", "ArrowRight", "ArrowUp", "ArrowDown",
+ * "Backspace", "Delete", "Home", "End", "PageUp", "PageDown".
+ *
+ * The rest are modifiers that when absent mean don't care (either state -- true or false --
+ * matches), and when present require that modifier state to match. All present modifers must
+ * match for the event to pass the filter.
+ *
+ * @platform macos
+ */
+export type HandledKey = $ReadOnly<{|
+  key: string,
+  capsLock?: ?boolean,
+  shift?: ?boolean,
+  ctrl?: ?boolean,
+  alt?: ?boolean,
+  meta?: ?boolean,
+  numericPad?: ?boolean,
+  help?: ?boolean,
+  function?: ?boolean,
+|}>;
+
+export type KeyboardEventProps = $ReadOnly<{|
+  /**
+   * Called after a key down event is detected.
+   */
   onKeyDown?: ?(event: KeyEvent) => void,
+
+  /**
+   * Called after a key up event is detected.
+   */
   onKeyUp?: ?(event: KeyEvent) => void,
+
+  /**
+   * When `true`, allows `onKeyDown` and `onKeyUp` to receive events not specified in
+   * `validKeysDown` and `validKeysUp`, respectively. Events matching `validKeysDown` and `validKeysUp`
+   * are still removed from the event queue, but the others are not.
+   *
+   * @platform macos
+   */
+  passthroughAllKeyEvents?: ?boolean,
```

Comment:
It was suggested to me that instead of a prop, we could add a feature flag, which both React Native and React Native Windows utilize to an extent: https://github.com/microsoft/react-native-macos/blob/main/Libraries/ReactNative/ReactNativeFeatureFlags.js

That would gate the behavior to a single instance of React Native, which might work for our case.

Reply:
I ended up trying that out over here: https://github.com/microsoft/react-native-macos/pull/1615

Reply:
Talked offline, we'll keep it a prop for now. I don't want to overcomplicate this change